docs/requirements.md

```diff
@@ -2,78 +2,48 @@
+-   A Kubernetes 1.14.\* or 1.15.\* cluster provisioned with the [required resources](#hardware-requirements)
+-   The [NGINX Ingress Controller](https://github.com/helm/charts/tree/master/stable/nginx-ingress) must be setup in your cluster
 -   A [PostgreSQL server](#postgresql-server-setup) accessible from the Kubernetes cluster
 
-## Cluster resource requirements
+## Hardware Requirements
 
-To have an enjoyable experience with Codacy, you should have the
-following calculations in mind when allocating resources for the
-installation and defining the number of concurrent analysis.
+A good way of understanding how resources are used in Codacy is by understanding how the resources are allocated.
 
-Without accounting for analysis ([next section](#analysis)),
-you should need at least the following resources:
+!["High Level Architecture"](<./images/High Level Architecture - Analysis II.svg> "High Level Architecture")
 
-```text
-CPU: 7 CPU
-Memory: 40 GB
-```
+You can look at Codacy separately as the "Platform" and the "Analysis" cluster. The Platform contains the UI and other
+components important to treat and show results. The Analysis is the swarm of workers that run several linters for
+your project.
 
-Check the
-[values-production.yaml](https://github.com/codacy/chart/blob/master/codacy/values-production.yaml)
-file to find a configuration reference that should work for you to run a
-"production-ready" version of Codacy.
+**The resources needed for Codacy depend a lot on the rate of commits done by your team.**
 
-### Analysis
+Since all of this runs in kubernetes, you can increase the number of replicas in everything, which should give you
+more resilience and throughput, but it will also increase resource usage. A very simplistic overview of the resource
+allocations for each of these is something like this:
 
-Each analysis runs a maximum number of 4 plugins in parallel (not configurable)
+| Component                           | vCPU | Memory |
+| ----------------------------------- | ---- | ------ |
+| Platform w/ 1 replica on everything | 4    | 8      |
+| Per Worker                          | 5    | 10     |
 
-Note: All the following configurations are nested inside the `worker-manager.config`
-configuration object, but for simplicity, we decided to omit the full path.
+The following table resources are based on our experience and the values we allocated for
+resources in the [values-production.yaml](https://raw.githubusercontent.com/codacy/chart/master/codacy/values-production.yaml)
+and you might need to adapt them to your use case.
 
-```text
-CPU: workerResources.requests.cpu + (pluginResources.requests.cpu * 4)
+| Installation type                        | Replicas per component | Max. commits analyzed concurrently | Platform vCPUs | Platform Memory | Analysis Workers vCPUs | Analysis Workers Memory | ~ Total vCPUs | ~ Total Memory |
```

Comment:
We should figure out a way of highlighting the **total** vCPUs and memory needed.

A simple way of doing this could be moving these two columns to the beginning of the table, or maybe we could use some other way of doing this.

Reply:
I changed these columns to the beginning... what do you think?

Reply:
Yes, it's better now because the more important information is readily visible at the beginning of the table.

I was thinking if it's really important to detail the vCPUs and RAM for the platform and the analysis workers.